app/controllers/exercises_controller.rb

```diff
@@ -93,6 +97,17 @@ def edit
     collect_set_and_unset_exercise_tags
   end
 
+  def push_proforma_xml
+    codeharbor_link = CodeHarborLink.find(params[:account_link])
+    oauth2Client = OAuth2::Client.new('client_id', 'client_secret', :site => codeharbor_link.push_url)
```

Comment:
client_id and client_secret will be set correctly in the next merge_request.
We will adjust the procedure on CodeHarbor accordingly. 
A group admin will be able to register an autograder with CodeHarbor.
The groups on CodeHarbor will be extended with a third role: instructor

- admins can add and edit users, exercises and autograders to the group.  
- members can view the exercises and export them to xml
- instructors can do the same as members plus pushing them to the registered autograder.

Reply:
this has not yet happened.